tests/main/test_main_general.py

```
@@ -2466,3 +2466,51 @@ def test_config_models_allow_internal_model_extra_options() -> None:
+    )
+    assert_output(result, EXPECTED_MAIN_PATH / "generate_typeddict_with_datetime_class_uses_datetime.py")
+
+
+def test_generate_typeddict_with_non_datetime_class_raises_error() -> None:
+    """Test that TypedDict rejects pydantic-specific datetime types with a clear error."""
+    from datamodel_code_generator.format import DatetimeClassType
+
+    json_schema = '{"type": "object", "properties": {"created_at": {"type": "string", "format": "date-time"}}}'
+    with pytest.raises(
+        Error,
+        match='`--output-datetime-class` only allows "datetime" for `--output-model-type` typing.TypedDict',
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)